tests/tcpclientsrc.rs

```diff
@@ -0,0 +1,152 @@
+        }
+        #[cfg(not(debug_assertions))]
+        {
+            use std::path::Path;
+
+            let mut path = Path::new("target/release");
+            if !path.exists() {
+                path = Path::new("../target/release");
+            }
+
+            gst::Registry::get().scan_path(path);
+        }
+    });
+}
+
+#[test]
+fn test_push() {
+    init();
+
+    let handler = thread::spawn(move || {
+        use std::net;
+
+        let listener = net::TcpListener::bind("0.0.0.0:5000").unwrap();
+        for stream in listener.incoming() {
+            let buffer = [0; 160];
+            let mut socket = stream.unwrap();
+            for _ in 0..3 {
+                let _ = socket.write(&buffer);
+                thread::sleep(time::Duration::from_millis(20));
+            }
+            break;
+        }
+    });
+
+    let pipeline = gst::Pipeline::new(None);
+
+    let tcpclientsrc = gst::ElementFactory::make("ts-tcpclientsrc", None).unwrap();
+    let appsink = gst::ElementFactory::make("appsink", None).unwrap();
+    appsink.set_property("sync", &false).unwrap();
+    appsink.set_property("async", &false).unwrap();
+
+    pipeline.add_many(&[&tcpclientsrc, &appsink]).unwrap();
+    tcpclientsrc.link(&appsink).unwrap();
+
+    let caps = gst::Caps::new_simple("foo/bar", &[]);
+    tcpclientsrc.set_property("caps", &caps).unwrap();
+    tcpclientsrc.set_property("port", &(5000u32)).unwrap();
+
+    appsink.set_property("emit-signals", &true).unwrap();
+
+    let samples = Arc::new(Mutex::new(Vec::new()));
+
+    let samples_clone = samples.clone();
+    appsink
+        .connect("new-sample", true, move |args| {
+            let appsink = args[0].get::<gst::Element>().unwrap();
+
+            let sample = appsink
+                .emit("pull-sample", &[])
+                .unwrap()
+                .unwrap()
+                .get::<gst::Sample>()
+                .unwrap();
+
+            let mut samples = samples_clone.lock().unwrap();
+
+            samples.push(sample);
+            if samples.len() == 3 {
+                let _ = appsink.post_message(&gst::Message::new_eos().src(Some(&appsink)).build());
+            }
+
+            Some(gst::FlowReturn::Ok.to_value())
+        })
+        .unwrap();
+
+    pipeline
+        .set_state(gst::State::Playing)
+        .into_result()
+        .unwrap();
+
+    let mut eos = false;
+    let bus = pipeline.get_bus().unwrap();
+    while let Some(msg) = bus.timed_pop(5 * gst::SECOND) {
+        use gst::MessageView;
+        match msg.view() {
+            MessageView::Eos(..) => {
+                eos = true;
+                break;
+            }
+            MessageView::Error(..) => unreachable!(),
+            _ => (),
+        }
+    }
+
+    assert!(eos);
+    let samples = samples.lock().unwrap();
+    assert_eq!(samples.len(), 3);
+
+    for sample in samples.iter() {
+        assert_eq!(sample.get_buffer().map(|b| b.get_size()), Some(160));
```

Comment:
And same here, we can't know anything about the number and size of buffers. We only know that overall we will have `3*160` bytes

Reply:
Fixed it 8b1dd1a